Macros/Macros.js

```diff
@@ -5,20 +5,22 @@ class FurnaceMacros {
                 const macro = game.macros.entities.find(macro => macro.name === name);
                 if (!macro) return "";
                 const result = macro.renderContent(...args);
-                if (typeof(result) !== "string")
+                if (typeof (result) !== "string")
                     return "";
                 return result;
             }
         }
         Handlebars.registerHelper(helpers)
-        FurnaceMacros._moduleSocket = "module.furnace";
 
-        this._setupSocket();
+        this._GMElectionIds = [];
```

Comment:
I like the election idea.
Not sure I understand the aversion to Hooks (not that I have a strong view) haven't you more or less implemented it again and called it requestResolvers? The issue of name collision aside that is.

Reply:
Well, hooks are "core/module sending a notification to everyone else to act upon something that happens". This is all internal to Furnace so it shouldn't be sent as a hook. Just like the socket event is named 'module.furnace' and not "everyone".
Also, I always have debugging on, so I see the hooks called in the console all the time to see what core is doing, and having that appear in there somehow feels wrong :p